cpp/oneapi/dal/algo/louvain/test/louvain_test.cpp

```diff
@@ -224,9 +224,12 @@ struct GraphTypeInfo<std::int32_t> {
 
 class louvain_test {
 public:
+    static bool is_nan(double x) {
+        return x != x;
+    }
```

Comment:
Are you sure this will work for all nan values?

Reply:
Yes. It works: https://en.wikipedia.org/wiki/NaN#Comparison_with_NaN